inc/ConfigFileParser.hpp

```diff
@@ -1,12 +1,61 @@
 #pragma once
 
+/* ====== LIBRARIES ====== */
+
 #include "ConfigFile.hpp"
+#include <algorithm>
+#include <fstream>
+#include <set>
+#include <stack>
+#include <string>
+#include <iostream>
+
+/* ====== DEFINITIONS ====== */
+
+#define SERVER 1
+#define LOCATION 2
+
+/* ====== CLASS DECLARATION ====== */
 
 class ConfigFileParser {
 public:
-    ConfigFileParser(std::string configFile);
-    ~ConfigFileParser();
+
+	const ConfigFile& parseConfigFile(const std::string& configFilePath);
+
+    ConfigFileParser(void);
 
 private:
+
+	ConfigFileParser(const ConfigFileParser& ref);
+	ConfigFileParser& operator=(const ConfigFileParser& ref);
```

Comment:
> I thought that I should them define them and set then both as private so they can't be used.

In this case I think it's more correct to just declare them in the header but not actually define them in the source file. This way it's not possible to use them (even in context of private usage) since they are not defined anywhere > compile error. 

Before I realized this I also wrote some "dummy" implementations of copy ctor & assignment since I thought you need to define them. I would get rid of them in some cleanup PR maybe.

But since you declare a copy ctor and assignment (even if it's just for disabling copying), you'd also should define a dtor. In newer standard you'd just write `~ConfigFileParser() = default`. Since we have to use old standard, a empty dtor does the same thing.

Reply:
Makes sense. 

I declared the copy ctor & assignment only in the header and removed their implementations from the cpp file.
Additionally I added a destructor which is declared in the header file and implemented in the cpp file.